runtime/moonbase/src/lib.rs

```diff
@@ -707,9 +707,22 @@ impl pallet_author_inherent::Config for Runtime {
 	type WeightInfo = pallet_author_inherent::weights::SubstrateWeight<Runtime>;
 }
 
+/// Uses pallet_randomness, falling back to pallet_randomness_collective_flip
+pub struct GuaranteedRandomness;
+impl frame_support::traits::Randomness<Hash, BlockNumber> for GuaranteedRandomness {
+	fn random(subject: &[u8]) -> (Hash, BlockNumber) {
+		let maybe_randomness = Randomness::random(subject.clone());
+		if maybe_randomness.0.is_some() {
+			return (maybe_randomness.0.unwrap(), maybe_randomness.1);
+		}
+
+		pallet_randomness_collective_flip::Pallet::<Runtime>::random(subject)
```

Comment:
collective flip is the fallback? I don't think this is necessary because **pallet-randomness will panic on `on-initialize` if the LocalVrfOutput is not set** so we know that it is set even though it is an option value

Reply:
That is true if we don't go for the delay approach. Elsewise, if we decide to use the `N` block delayed vrf, then at genesis we'd have the `N-1` starting blocks that will not have the vrf history in place. In this case we'd still need something.

Reply:
We don't need something, we just delay it for the first N blocks.